### DISCUSSION TITLE
Want to match aliasColors per panel resource even when using visualization "graph old" in v8.

### DISCUSSION THREAD

**Discussion by SEIJIIWATA** [grafana/grafana#44174] [Dashboards, Panels & Visualizations]
https://github.com/grafana/grafana/issues/41782
Is this problem expected to be resolved with grafana version up grade?
In which version was it fixed?

**Comment by SEIJIIWATA** [comment]
![2022-01-17 13 54のイメージ](https://user-images.githubusercontent.com/57309320/149710680-9e4c25a0-a635-46fd-932f-c443e048ca73.jpeg)

Even in the current latest version, v8.3.3, if the number of resources is different for each metric, the colors of the graphs displayed will vary.
This can lead to unintentional human error in case of failure.

**Comment by SEIJIIWATA** [comment]
@torkelo 
Can you confirm this?

**Comment by torkelo** [comment]
What do you mean? 

You can set the colors to what ever you want. If the number of metics change then new colors will be used for the series colors

**Comment by torkelo** [comment]
To get series returned by query in different orders by two panels then you need to manually set colors to sync them. The automatic color set based on the order the series are returned so if two panels have a query that return same series but with different order then the auto color will be different for the same series names

**Reply by SEIJIIWATA** [reply]
@torkelo -san
Thank you for providing a forum for discussion.

At least in v6.4.1, even if the number of graph resources was different in different metrics panels, the colors of the common graph resources were the same except for the difference graph resources.
After v8, the colors of the common graph resources also changed and became different.

Is it possible to share the code part that shows the change of aliasColors specification from v6.4.1 to v8?

**Reply by torkelo** [reply]
We have not changed the coloring logic , still based on series order 

So series 1 get color 1, series 2 get color 2, etc


Of course any manually set color will override the auto color.

This logic has been the same since v1 so no difference in v8 compared to v6

**Reply by torkelo** [reply]
I think what could cause the issue is that the query result order is not consistent and changes over time or between panels , this can cause different colors

**Reply by SEIJIIWATA** [reply]
I'll follow your words and test again with v6.

**Reply by SEIJIIWATA** [reply]
@torkelo -san
In v6, even if you specify All in the variable, graph resources that do not have a value in the metrics are also displayed, so there is no difference in the number of resources, so the colors do not shift?
![2022-01-19 20 55のイメージ](https://user-images.githubusercontent.com/57309320/150125531-5eef85b5-8b25-4eab-9f57-93a4a53cd478.jpeg)

**Comment by torkelo** [comment]
> In v6, even if you specify All in the variable, graph resources that do not have a value in the metrics are also displayed, so there is no difference in the number of resources, so the colors do not shift? 

what do you mean? colors shift? which ones? 

If the series order is different then two series with the same name will get different color. The auto color is only assigned by the order in which the data source returned series

**Reply by SEIJIIWATA** [reply]
@torkelo -san

In v6, there is no discrepancy in color assignment to resources in different metrics panels, and they are identical.
In v8, there are cases where there is a discrepancy in the color assignment for resources in different metrics panels.
For example, the first image is v8.3.3, but "i-0e81cc163c40e384a" in the "CPU utilization per instance [max %]" panel and "i-0e81cc163c40e384a" in the "CPUSurplusCreditBalance" panel is the colors assigned to them are clearly different, aren't they?

The reason for this, in my opinion, is that in v6, EC2 resources that do not have a value in the datasouce such as cloudwatch are displayed.
When I reverted back to v8.3.3, EC2 resources without values in the datasouce such as cloudwatch were not displayed.
In other words, I think that when comparing different metrics panels, if there is a difference in the number of resources displayed in each panel, the color assignment to the resources will be different.

Do you understand what I'm saying?

▼v8.3.3
![2022-01-20 13 18のイメージ](https://user-images.githubusercontent.com/57309320/150272998-0845b1ec-ed85-4070-bb31-92a25521ce0f.jpeg)

▼v6.4.1
![2022-01-19 20 55のイメージ](https://user-images.githubusercontent.com/57309320/150274481-80ada07f-a08b-423b-ba17-2ea7b9ce761f.jpeg)

**Comment by torkelo** [comment]
it could be due to cloudwatch not returning series in the same order anymore

**Reply by SEIJIIWATA** [reply]
@torkelo -san
I also believe that this could be one of the factors.
I believe the answer to this inquiry can be summed up in a nutshell as follows, and if it is correct, we can close this discussion.

・Cloudwatch not returning series in the same order anymore
・Since v8, it has been hidden for resources that have no value.

**Reply by SEIJIIWATA** [reply]
@torkelo -san
Since there seems to be no answer, I think I can close with the following understanding.

・Cloudwatch not returning series in the same order anymore
・Since v8, it has been hidden for resources that have no value.

**Comment by sarahzinger** [comment]
Hey folks,
As part of our work moving from [discussions to issues](https://github.com/grafana/grafana/discussions/73424) I'm going through some older discussions to see what we might be able to close. It seems like this one can be closed since it's a bit out of date, but if we've missed something feel free to open an issue and we can take another look!
